src/rt/cpp/cown_span.h

```diff
@@ -0,0 +1,67 @@
+// Copyright Microsoft and Project Verona Contributors.
+// SPDX-License-Identifier: MIT
+#pragma once
+
+namespace verona::cpp
+{
+
+  /**
+   * This is a span of cown_ptr that has ownership of the included cown_ptrs
+   *
+   * The default constructor takes an array of cown_ptr and heap allocates
+   * another array to hold those pointers after incrementing the reference
+   * count. Alternatively, the constructor takes a template argument to move
+   * the cown_ptr array and avoid the allocation.
+   * In both cases cown_ptr_span has ownership over the cown_ptr
+   *
+   * The destructor calls the destructor of each cown_ptr and frees the
+   * allocated array.
+   */
+  template<typename T>
+  struct cown_ptr_span
```

Comment:
Sorry, I have one minor thing given this is an owning array.  I think we should perhaps call it: `cown_array`.  I think of `span` as borrowed, and not owning the underlying memory.  But this does, so I think the current name will given the wrong idea.

Reply:
Sure I can do that. I thought that owning made more sense and made things simpler. Alternatively, we can keep the `span` semantics and prevent from passing it with `move`. What do you think?

Reply:
I think owning is the right choice. Let's rename.

Reply:
Ack